automower_ble/mower.py

```diff
@@ -111,23 +116,68 @@ async def mower_resume(self):
 
     async def mower_park(self):
         await self.set_parameter("park")
+    
+    async def send_keepalive(self):
+        await self.set_parameter("keepalive")
+        return True
+    
+    async def send_operator_pin_request(self, pin):
+        await self.set_parameter('pin', code=pin)
+        return True
+    
+    async def get_startupsequence_required_request(self)-> bool:
+        required = await self.get_parameter('getStartupSequenceRequiredRequest')
+        return required
+    
+    async def is_operator_loggedin(self):
+        result = await self.get_parameter('isOperatorLoggedIn')
+        return result
+    
+    async def get_serial_number(self):
+        sn = await self.get_parameter('serialNumber')
+        return sn
+    
+    async def get_restriction_reason(self):
+        rr = await self.get_parameter('getRestrictionReason')
+        if rr is None:
+            return None
+        return RestrictionReason(rr)
+    
+    async def get_number_of_tasks(self):
+        response = await self.get_parameter('getNumberOfTasks')
+        return response
+    
+    async def get_mode_of_operation(self):
+        mode_response = await self.get_parameter('getModeOfOperation')
+        if mode_response is None:
+            return None
+        return ModeOfOperation(mode_response)
+    
+    async def set_mode_of_operation(self, mode:ModeOfOperation):
+        await self.set_parameter('setModeOfOperation', mode=mode )
```

Comment:
```suggestion
        await self.set_parameter('setModeOfOperation', mode=mode.value )
```
The value of the Enum should be passed in. I now notice this should also be changed in `mower.mower_override` function.

Reply:
Updated code based on comments